File: templates/service-java/{{#server.micronaut}}/src/main/java/{{package.value}}/{{#jsonlib.jackson}}/ObjectMapperConfig.java
package {{package.value}};

import com.fasterxml.jackson.databind.ObjectMapper;
import io.micronaut.context.annotation.*;
import io.micronaut.jackson.ObjectMapperFactory;
import {{package.value}}.json.*;

@Factory
@Replaces(ObjectMapperFactory.class)
public class ObjectMapperConfig {
    @Bean
	@Replaces(ObjectMapper.class)
	public ObjectMapper objectMapper() {
		var objectMapper = new ObjectMapper();
		CustomObjectMapper.setup(objectMapper);
		return objectMapper;
    }

    @Bean
    public Json json() {
        return new Json(objectMapper());
    }
}

File: templates/service-kotlin/{{#server.micronaut}}/src/main/kotlin/{{package.value}}/{{#jsonlib.jackson}}/ObjectMapperConfig.kt
package {{package.value}}

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import io.micronaut.context.annotation.*
import io.micronaut.jackson.ObjectMapperFactory
import {{package.value}}.json.*

@Factory
@Replaces(ObjectMapperFactory::class)
class ObjectMapperConfig {
    @Bean
    @Replaces(ObjectMapper::class)
    fun objectMapper(): ObjectMapper {
        val objectMapper = jacksonObjectMapper()
        setupObjectMapper(objectMapper)
        return objectMapper
    }

    @Bean
    fun json(): Json {
        return Json(objectMapper())
    }
}